src/linearization.jl

```diff
@@ -0,0 +1,229 @@
+"""
+    LinearizedModel{M,L,T} <: AbstractModel
+
+A container for the linearized model that holds the full nonlinear model, the linearized model, and the
+trajectory of linearization points. The same dynamics and jacobian functions can still be called on the
+`LinearizedModel` type.
+"""
+struct LinearizedModel{M, L, T} <: AbstractModel
+    model::M
+    linmodel::L
+    trajectory::T
+end
+
+"""
+    LinearizedModel(model::AbstractModel, linmodel::LinearModel, traj::AbstractTrajectory)
+
+A constructor for the linearized model type. The inputted model should be the full continuous dynamics
+model to be linearized.
+"""
+LinearizedModel(model::M, linmodel::L, traj::T) where {M <: AbstractModel, L <: LinearModel, T <: AbstractTrajectory} = 
+    LinearizedModel{M, L, T}(model, linmodel, traj)
+
+
+dynamics(model::LinearizedModel, x, u, t=0.0) = dynamics(model.linmodel, x, u, t)
+discrete_dynamics(::Type{PassThrough}, model::LinearizedModel, x, u, t, dt) = 
+    discrete_dynamics(PassThrough, model.linmodel, x, u, t, dt)
+
+jacobian!(∇f::AbstractMatrix, model::LinearizedModel, z::AbstractKnotPoint) = 
+    jacobian!(∇f, model.linmodel, z)
+
+discrete_jacobian!(::Type{PassThrough}, ∇f, model::LinearizedModel, z::AbstractKnotPoint) = 
+    discrete_jacobian!(PassThrough, ∇f, model.linmodel, z)
+
+"""
+    linearize_and_discretize(::Type{Q}, nonlinear_model::AbstractModel, trajectory::AbstractTrajectory) where {Q<:Explicit}
+
+Linearize `nonlinear_model` about `trajectory` and discretize with given integration type `Q`. Returns a [`LinearizedModel`](@ref) 
+"""
+function linearize_and_discretize(::Type{Q}, nonlinear_model::AbstractModel, trajectory::AbstractTrajectory; is_affine=false) where {Q<:Explicit}
```

Comment:
Can we simplify these methods to an interface similar to `LinearModel`? I think it would make sense to have a powerful `LinearizedModel` constructor that handles continuous/discrete, affine/non-affine, and then have an `update_reference!(model, Z)` that re-linearized the model about the updated reference trajectory `Z` (could be a single `KnotPoint`).